hushline/model.py

```diff
@@ -39,72 +75,129 @@ class User(Model):
     )
 
     @property
-    def password_hash(self) -> str:
+    def _aad_secret(self) -> bytearray:
+        domain = b"user_aad_secret"
+        aad = deque([self.__tablename__.encode()])
+        vault = current_app.config["VAULT"]
+        return bytearray(vault.decrypt(self._hidden_aad_secret, domain=domain, aad=aad))
+
+    @_aad_secret.setter
+    def _aad_secret(self, value: bytes) -> None:
+        domain = b"user_aad_secret"
+        aad = deque([self.__tablename__.encode()])
+        vault = current_app.config["VAULT"]
+        self._hidden_aad_secret = vault.encrypt(value, domain=domain, aad=aad)
+
+    @property
+    def password_hash(self) -> str | None:
         """Return the hashed password."""
         return self._password_hash
 
     @password_hash.setter
     def password_hash(self, plaintext_password: str) -> None:
-        """Hash plaintext password using scrypt and store it."""
-        self._password_hash = scrypt.hash(plaintext_password)
+        """Hash plaintext password using argon2 and store it."""
+        domain = b"user_password_hash"
+        vault = current_app.config["VAULT"]
+        self._password_revision_number += 1
+        with temp_user_aad(self) as aad:
+            aad.append(self._password_revision_number.to_bytes(8, "big"))
+            aad.append(bytearray(plaintext_password, encoding="utf-8"))
```

Comment:
The encodings here aren't committed to. Also, the revision number could be bigger to allow for algorithm version information to be encoded into a prefix.

Reply:
Addressed by (https://github.com/scidsg/hushline/pull/411/commits/c23c9beb78af1265ec52f56efa9c0740759c7724)